tensorflow_encrypted/protocol/pond.py

```diff
@@ -878,7 +896,9 @@ def __init__(
     ) -> None:
         assert isinstance(value_on_0, AbstractTensor), type(value_on_0)
         assert isinstance(value_on_1, AbstractTensor), type(value_on_1)
-        assert value_on_0.shape == value_on_1.shape
+
+        # as list is performed to compare partially unknown shapes (e.g. (?, 1) & (?, 1))
+        assert value_on_0.shape.as_list() == value_on_1.shape.as_list()
```

Comment:
pretty sure @mortendahl  had some reasons for the shape needing to be always defined

Reply:
uh oh .. it is never known out of equal

Reply:
Yeah that makes sense. I think the potential problems will be caused by `Int100Tensor` not 100% sure... Maybe we just need to deal with it if we need to use `tf.equal`.

Reply:
yeah we need the shape to always be known, ie. `x.shape.is_fully_defined()` must be true. what do you mean by "it is never known out of equal"?

Reply:
sorry, I mean out of `where` .. e.g.
```python
a = tf.constant(np.array([1,0,1,0]))
eq = tf.equal(a, 0)
w = tf.where(eq)
print(w)
```
will print

```
Tensor("Where:0", shape=(?, 1), dtype=int64)
```

Reply:
[the doc](https://www.tensorflow.org/api_docs/python/tf/where) says "Returns:
A Tensor with the same type and shape as x, y if they are non-None", so what if you do `tf.reshape(w, a.shape)` afterwards?